### PR TITLE
Refactor to be able to document deleted message

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Refactor before #6670

### CODE REVIEW

File: doc/exts/pylint_messages.py [hunk 1]
```diff
@@ -60,61 +60,72 @@ def _register_all_checkers_and_extensions(linter: PyLinter) -> None:
...
-{file_content_string}"""
-    return good_code, bad_code, details, related
+def _create_placeholders(
+    data_path: Path, details_rst_path: Path, good_py_path: Path
+) -> None:
+    data_path.mkdir(parents=True)
+    with open(good_py_path, "w", encoding="utf-8") as file:
+        file.write(
+            """\
+# This is a placeholder for correct code for this message.
```
Comment (DanielNoord) [inline doc/exts/pylint_messages.py:120]: Shall we remove those other files then?

Comment (Pierre-Sassoulas) [reply doc/exts/pylint_messages.py:120]: What files are we talking about ?

Comment (DanielNoord) [reply doc/exts/pylint_messages.py:120]: The files you added recently in the placeholder PR.

Comment (Pierre-Sassoulas) [reply doc/exts/pylint_messages.py:120]: Ha, right. I was thinking I added the function at the same time than the files but it was not the case. Well I think it's a good thinq that we can generate easily but also that a new contributor can just use the already generated file when he's doing the doc addition, so we might keep both.

Comment (DanielNoord) [reply doc/exts/pylint_messages.py:120]: Yeah the only thing I worried about is that we might forget to remove one of the placeholder files if we update the other, as the placeholder text is divided over two different files.

Comment (Pierre-Sassoulas) [reply doc/exts/pylint_messages.py:120]: Yeah this is a little dangerous if the contributor forget to remove the details.rst and there's no detail the review seems fine at first glance but it's not.

Comment (Pierre-Sassoulas) [reply doc/exts/pylint_messages.py:123]: I added an assert

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**